web/src/giphy.ts

```diff
@@ -10,30 +16,35 @@ import * as rows from "./rows";
 import {realm} from "./state_data";
 import * as ui_util from "./ui_util";
 
-let giphy_fetch;
+let giphy_fetch: GiphyFetch | undefined;
 let search_term = "";
-let gifs_grid;
-let giphy_popover_instance;
+let gifs_grid: {remove: () => void} | undefined;
+let giphy_popover_instance: tippy.Instance | undefined;
 
 // Only used if popover called from edit message, otherwise it is `undefined`.
-let edit_message_id;
+let edit_message_id: number | undefined;
 
-export function is_popped_from_edit_message() {
+const rating_schema = z.enum(["pg", "g", "y", "pg-13", "r"]);
```

Comment:
IMO it would be a little more readable to shift this `z` const above the local `var` variables.